### ISSUE TITLE
The app keeps crashing

### ISSUE THREAD

**Issue by en-coded** [Azure-Samples/azure-search-openai-demo#504]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> The app crashes after a user asks 4-5 questions. 

I am looking for suggestions to debug the application crashing issue and if anyone else is facing the problem of running this application in a production-like environment. Does this demo app provide a scalable solution?

**Here is our current configuration:**
App Backend is on App Service Plan P2V3 with three instances with health check.
Search Service is on three replicas.

We have less than ten users trying this application, but it cannot support the basic workload despite upgrading our backend and search services.

I have contacted Azure support with a critical support request, but the response has been very slow. A few solutions have been proposed, but nothing seems to work.

As of today, I am getting a new error Error: HTTP code 502 from API (). There is not much information in the console log - just the error code.

### Any log messages given by the failure
>  HTTP code 502 from API ()

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
All operating systems

### azd version?
> run `azd version` and copy paste here.
azd version 1.1.0
### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
Have you tried my Quart branch? https://github.com/Azure-Samples/azure-search-openai-demo/pull/503
If your app is crashing due to thread contention causing timeouts, the asynchronous approach should be more robust. I am currently loadtesting it with a load of 50 concurrent users.

**Comment by en-coded** [comment]
No I have not. I will give it a shot and report back. Thanks.

**Comment by vrajroutu** [comment]
@pamelafox could you also merge the streaming feature to the quart branch? Streaming is much more faster and starts generating the responses right away, instead of waiting for the complete response to be generated.

**Comment by vrajroutu** [comment]
@en-coded Azure OAI Outage in EAST-US effecting emedding models

FYI

Your support case @####$#%$^#&%&$&&$ is related to an ongoing outage in your region.

STATUS: In-Progress 8/8/2023 4:43:29 AM UTC

SUMMARY OF IMPACT: You may experience sporadic issues accessing your resources. This could cause you to experience a 5xx error when calling Davinci model or Ada embedding model in EastUS region.

NEXT STEPS: Azure OpenAI feature team are urgently investigating the root cause and coming up with a plan to resolve as soon as possible. We'll send an update once we know more.

**Comment by en-coded** [comment]
Thanks @vrajroutu! That at least explains the error HTTP code 502 from API (). I was ready to rip apart everything otherwise :)

Hoping to get a fix for app performance via #503

**Comment by en-coded** [comment]
I implemented the changes as per https://github.com/Azure-Samples/azure-search-openai-demo/pull/503, and it has helped significantly with eliminating the app crashes. 

The query response is still slow, but the apps are not crashing since I implemented the fix. 

Thank you, @pamelafox and @vrajroutu.
